### ISSUE TITLE
[Improvement] Add expensify.cash to iOS and Android native share menus

### ISSUE THREAD

**Issue by sakluger** [Expensify/App#1294]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

**Platform - version:**  
iOS, Android - All versions

## Problem

A common action when using a mobile device is to share something (a web page, an image, a tweet, etc) with another person. Both iOS and Android have a native "share" button that allows you to share things via apps that you have installed on your phone.

Here's an example of what the native iOS share menu looks like: 
![image](https://user-images.githubusercontent.com/1371996/103492514-0fa71f80-4df1-11eb-9eb0-b803669edc9d.png)

## Why is this important?
The native share function is a common flow and one of the easier ways to share something with someone. For example, I find it to be easier than copying the url of a web page from my mobile browser, switching apps, and pasting it in to the messenger app. It's much easier to just hit the share button, choose the messenger app, and select who I want to send the url to. 

## Solution
Add support for sharing directly to our app. Here is an [example react native library](https://github.com/meedan/react-native-share-menu) that helps with this, though I haven't done any research into it. I think the way we would implement it is: 
1. Expensify.cash shows up as an option in the native share menu
2. When Expensify.cash is selected, we bring up your Expensify.cash home screen (Screenshot is below)
3. You can select a chat 
4. Whatever you are sharing (url, image, etc) is sent to that chat

This is very similar to the flow when using the native share functionality with WhatsApp. 

Screenshot of Expensify.cash home screen:

![image](https://user-images.githubusercontent.com/5997400/105258369-7bea8880-5b3e-11eb-9949-7f6a9c315784.png)

Upwork Job Link: https://www.upwork.com/jobs/~01fb534e5bfc655563

**Comment by tugbadogan** [comment]
Is this task still available? I would like to work on this issue.

**Comment by sakluger** [comment]
Hi @tugbadogan thanks for the interest! It is not available, but any available issues should be posted on Upwork.

**Comment by r8** [comment]
I evaluated several React Native libraries, but it looks like the real choice is only between two of them:
- [react-native-share-extension](https://github.com/alinz/react-native-share-extension)
- [react-native-share-menu](https://github.com/meedan/react-native-share-menu)

The former looks more popular, but requires manual install and haven't been updated in a while, which causes a lot of troubles and compatibility issues with latest React versions. So probably it's better to go with react-native-share-menu, like it was proposed in issue description.

I've tested the integration, it works more or less fine, but has one issue on iOS: https://github.com/meedan/react-native-share-menu/issues/82

My proposal is to create `libs/ShareManager` module, similar to `ActiveClientManager` or `Notification`.
Something like that:

**libs/ShareManager/index.native.js**
```javascript
import ShareMenu from 'react-native-share-menu';

let listener;

function handleShare(item) {
    // Check the shared item, save it to Onyx 
    // and redirect to chat chooser interface
}

function register() {
    ShareMenu.getInitialShare(handleShare);
    listener = ShareMenu.addNewShareListener(handleShare);
}

function unregister() {
    listener.remove();
}

export {
    register,
    unregister,
};
```
The code should go `index.native.js`, and nooped in `index.js`, because it will not work in web and desktop anyway.
Then we should register in `componentDidMount` of `Expensify.js` and unregister in `componentWillUnmount`.

Instead of showing the Home screen with a list of chats I suggest to create a separate screen for that.
If we will reuse Home screen, we would have to:

- Hide user avatar, because when user taps on it, it will open the Settings screen and break the sharing flow
- Change the onTap callback for chats in the list, because instead of just opening the chat, first we would need to post a message.

With all these new conditions, checks (and the fact that sharing should work only on mobile) Home screen and Sidebar can become too heavy. So maybe it's better to create a separate `SharingScreen` and reuse sidebar components or even whole `SidebarView` there.
